tests/run/legacy_implicit_noexcept.pyx

```diff
@@ -51,46 +51,50 @@ def func_pure_implicit() -> cython.int:
 def func_pure_noexcept() -> cython.int:
     raise RuntimeError
 
-def print_stderr(func):
+def return_stderr(func):
     @functools.wraps(func)
     def testfunc():
-        from contextlib import redirect_stderr
-        with redirect_stderr(sys.stdout):
+        old_stderr = sys.stderr
+        stderr = sys.stderr = StringIO()
+        try:
             func()
+        finally:
+            sys.stderr = old_stderr
+        return stderr.getvalue().strip()
```

Comment:
```suggestion
        return print(stderr.getvalue())
```

I was wondering about doing this, which I think would let us leave everything else unchanged.